### DISCUSSION TITLE
Positions from TextMeshMSDF are wrong

### DISCUSSION THREAD

**Discussion by strangerattractor** [cables-gl/cables_docs#540] [! Welcome - readme first!]
Test Patch here: 
https://cables.gl/edit/i7cekz

I split the string into individual letters and use the positions from textMeshMSDF to position them with COPY and Transforms. 
But the letters don't line up. 
Why?

<img width="488" alt="image" src="https://github.com/cables-gl/cables_docs/assets/52496070/34661003-4217-431e-a9b7-ee3dffccc0fe">

Red Text is the normal TextMeshMSDF
White is taken from the positons.

**Comment by pandrr** [comment]
they are not wrong, font positioning is a complex thing, msdf fonts have positions and then you apply an offset then advancing etc etc etc... 

but in cables there is an easy way:
to animate textmeshmsdf it has a feedback input port for offsetting positions etc.
see this example: https://cables.gl/edit/g67Uo6
this way it also uses the internal mesh instancing and its only one drawcall for long texts

**Comment by strangerattractor** [comment]
Ah!!! Was also thinking it might be that.
Then I wanted to use a monospaced font to get around that issue.
But the Msdf also seams to „drop“ some signs or letters. I can’t really see
which.

The array of positions of letters is shorter then the number of letters I
feed in.

Could you check that too?

tom ***@***.***> schrieb am Mo. 19. Juni 2023 um 18:38:

> they are not wrong, font positioning is a complex thing, msdf fonts have
> positions and then you apply an offset then advancing etc etc etc... (see
> https://www.angelcode.com/products/bmfont/doc/file_format.html)
>
> but in cables there is an easy way:
> to animate textmeshmsdf it has a feedback input port for offsetting
> positions etc.
> see this example: https://cables.gl/edit/g67Uo6
> this way it also uses the internal mesh instancing and its only one
> drawcall for long texts
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/cables-gl/cables_docs/discussions/540#discussioncomment-6221586>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMQQNRTLC3ZWSVHSWSB6PWLXMB6CDANCNFSM6AAAAAAZMFHFC4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
*Max Schweder*
*www.maxschweder.de <http://www.maxschweder.de>*
*0176-66826992*

**Comment by strangerattractor** [comment]
I checked again and it seams that linebreaks are leading to a miscount of characters in the for loop. 
Linebreaks do not count as characters. Thus the count is of in the position output. 

there is 6 lines of text
in the last word, the last 6 letters all have the same position, because the position array is shorter by the amount of linebreaks.
